### ISSUE TITLE
datetime.strptime with Thai text supported

### ISSUE THREAD

**Issue by PhakphumV** [PyThaiNLP/pythainlp#766]
## Detailed description
With datetime.strptime, we can parse a string into a ```datetime``` object given a corresponding format.
However, this method doesn't support when a string is in Thai language.

## Context
Many open data available from Thai corporations or government are using date in Thai language such as ```14 มี.ค. 2549``` which may hinder data processing and analysis to have convert them manually.
By having this feature available, this could help to normalize Thai data easier regardless of which format each data source filing their date and time.

For me, I'm currently working on an open source jupyter notebook to extract historical interest rate from [bot.or.th](bot.or.th) and found that they are using Thai date format in their CSV file. You may find my repository from the link [here](https://github.com/PhakphumV/th-bank-historical-interest-rates). Once it's complete, it could be used for data visualization and further analysis.

## Possible implementation
I noticed that we already have some materials that we could use for this feature in ./pythainlp/util/date.py
But I'm quite new to this package so not sure if it's a good place to start with.

## Your environment
* PyThaiNLP version: 3.1.1
* Python version: 3.11

**Comment by wannaphong** [comment]
Hello! The [pythainlp.util.thai_strftime](https://pythainlp.github.io/dev-docs/api/util.html#pythainlp.util.thai_strftime) support only datetime type. It is not support string type.

**Comment by wannaphong** [comment]
Can you explain more about the function that you want to use?

**Comment by PhakphumV** [comment]
Hello, @wannaphong

The function [pythainlp.util.thai_strftime](https://pythainlp.github.io/dev-docs/api/util.html#pythainlp.util.thai_strftime) is to convert from datetime type to string type in a given format.

E.g. 
```
dt = datetime.date.today()
print(pythainlp.util.thai_strftime(dt,"%a %0d %b %y"))
```

Output will be
```
จ 16 ม.ค. 64
```

While the expected function is opposite, convert from string type to datetime type and having format directives as input to identify which part is date/month/year.

For example, the function named "thai_strptime"
```
dt_str = "อา 09 มิ.ย. 62"
dt = thai_strptime(dt_str,"%a %0d %b %y")
print(dt)
print(type(dt))
```

Output will be

```
2019-06-09
<class 'datetime.date'>
```

I noticed you're working on it in #767 and it looks good already (I tested your function too).
Looking forward for the next release. 
Thank you very much for your prompt reply and quick resolution.
Hope it will benefit other users too!
